### PR TITLE
Add initial e2e testing

### PR DESCRIPTION
**Pull request by mrogers950**
Adds rudimentary tests. After merging I'll add e2e runs to the openshift/release CI configs, and then follow up with more tests.
/cc @openshift/sig-security

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	if !hasPodWithPrefixName(adminClient, caControllerPodPrefix, serviceCAControllerNamespace) {
+		t.Fatalf("%s not running in %s namespace", caControllerPodPrefix, serviceCAControllerNamespace)
+	}
+
+	// test the main feature. annotate service -> created secret
+	t.Run("serving-cert-annotation", func(t *testing.T) {
+		ns := createTestNamespace(adminClient)
+		if len(ns) == 0 {
+			t.Fatalf("could not create test namespace")
+		}
```
Comment (ericavonb) [inline test/e2e/e2e_test.go]: Should we clean up the namespace as well?

Comment (mrogers950) [reply test/e2e/e2e_test.go]: we do in cleanupServiceSignerTestObjects()

Comment (ericavonb) [reply test/e2e/e2e_test.go]: What about in the situation that `createServinCertAnnotatedService` fails below?

Comment (mrogers950) [reply test/e2e/e2e_test.go]: Good catch, fixed that.

**Comment by mrogers950** [comment]
@ericavonb updated.

**Comment by smarterclayton** [comment]
/test e2e-aws

(this is the gating job for your operator to ensure you don't break the install or other components)

I would recommend that you create at least 1 e2e test inside of the origin suite (the conformance suite) that verifies that service service certs are working correctly, if you haven't already.  that's your smoke test to ensure other people don't break you.  you are welcome to put additional tests there, since that is where tests that validate an openshift cluster is providing the required function go.  if you'd like to do more stress testing of your operator, that belongs here (in this repo).

**Pull request closed without merging**